Advanced Tutorial.md

```diff
@@ -0,0 +1,211 @@
+              "type": "array",
+              "items": {
+                "type": "object",
+                "required": [
+                  "question", "published_at", "url", "choices"
+                ],
+                "properties": {
+                  "question": {
+                    "type": "string"
+                  },
+                  "published_at": {
+                    "type": "string"
+                  },
+                  "url": {
+                    "type": "string"
+                  },
+                  "choices": {
+                    "type": "array",
+                    "items": {
+                      "type": "object",
+                      "required": [
+                        "choice", "url", "votes"
+                      ],
+                      "properties": {
+                        "choice": {
+                          "type": "string"
+                        },
+                        "url": {
+                          "type": "string"
+                        },
+                        "votes": {
+                          "type": "number"
+                        }
+                      }
+                    }
+                  }
+                }
+              }
+            }
+```
+
+## Attributes
+
+Another way of describing examples and the structure of your request and response content is by using [MSON](https://github.com/apiaryio/mson#readme). MSON, like API Blueprint, allows you to use human-readable plain text to describe things rather than formats designed for computer parsing like JSON or YAML. Where API Blueprint allows you to describe your API, MSON allows you to describe data structure.
+
+MSON can be added to resources, actions, and individual requests or responses via an `+ Attributes` section. The random number generator API from above could look like this with MSON:
+
+```apib
+## Random Numbers [/random]
+
+### Generate a random number [GET]
+Generates and returns a random number.
+
++ Response 200 (application/json)
+
+    + Attributes
+
+        + value: 12345 (number)
+```
+
+When the above blueprint is parsed it will have a JSON body and JSON Schema example generated for it from the MSON attributes.
+
+The polls API question resource can also be modeled using MSON. By default, the `Attributes` section is assumed to be an object, but in this case we explicitly list it as an `array` instead:
+
+```apib
+### List All Questions [GET]
++ Response 200 (application/json)
+
+    + Attributes (array)
+
+        + (object)
+            + question: Favourite programming language? (required)
+            + published_at: `2014-11-11T08:40:51.620Z` (required)
+            + url: '/questions/1' (required)
+            + choices (array, required)
+                + (object)
+                    + choice: 'Javascript' (required)
+                    + url: '/questions/1/choices/1' (required)
+                    + votes: 2048 (number, required)
```

Comment:
It is an array of choices with a single sample. It's not entirely clear to me how to add more samples when you have an array of objects. Do you know how?